devtools/maven/src/main/java/io/quarkus/maven/BuildMojo.java

```diff
@@ -30,7 +34,7 @@
 /**
  * Builds the Quarkus application.
  */
-@Mojo(name = "build", defaultPhase = LifecyclePhase.PACKAGE, requiresDependencyResolution = ResolutionScope.COMPILE_PLUS_RUNTIME, threadSafe = true)
```

Comment:
We mark it as non thread safe as long as the problem with the quarkus profile (set as a System property) is not fixed

Reply:
Just curious, have you actually verified the system property set in one Mojo is visible in another one? You haven't I will check that.

Reply:
You can certainly re-check this but in the past I have filed multiple issues for various maven plugins that fell in the same trap, e.g. NPEs during iterating the properties while another Maven thread added or removed a property.

Reply:
Was that project properties of system properties? Thanks for the info @famod

Reply:
System properties, AFAIR.

Reply:
@aloubyansky yes I confirm that I have tested with a project with two modules, in each module I have 2 executions, in the first module, the first execution sets the Quarkus profile to `foo` and the second one sets the Quarkus profile to `bar` and in the second module, it does the opposite (first `bar` and second `foo`). I launched ` mvn -T 1C clean install` and ended up with a build success but incorrect Quarkus profile set at runtime (Quarkus profile set to `bar` instead of `foo` and vice versa). 

In other words, I confirm that system properties make the plugin non thread safe unless you use the same system properties across all your modules.

Reply:
Thanks a lot @essobedo

Reply:
@aloubyansky If you want to keep it thread safe, we can try to investigate more to see how we can isolate the Quarkus profile (by relying on the ClassLoader for example) but it won't be easy